langchain/prompts/chat.py

```diff
@@ -19,11 +19,11 @@
 
 
 class BaseMessagePromptTemplate(BaseModel, ABC):
-    prompt: BasePromptTemplate
+    prompt: BaseStringPromptTemplate
```

Comment:
is there any reason this needs to be string?

Reply:
I don't think this would work well with a ChatPromptTemplate, I mean technically you `format_prompt(**kwargs).to_string()` and set that as the `content` parameter of a message, but that seems really weird to me. Best to restrict these as string prompt templates